internal/kafka/internal/services/clusters.go

```diff
@@ -108,6 +110,21 @@ func (c clusterService) RegisterClusterJob(clusterRequest *api.Cluster) *apiErro
 	return nil
 }
 
+// DeregisterClusterJob deregister an existing cluster
+func (c clusterService) DeregisterClusterJob(clusterId string) *apiErrors.ServiceError {
+	cluster, err := c.FindClusterByID(clusterId)
+	if err != nil {
+		return err
+	}
+
```

Comment:
This is not needed.

Reply:
See https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1511#discussion_r1085017236 for suggestion

Reply:
What about the case where the cluster doesn't exist? I guess it's ok because then the update would fail?

Reply:
It depends on the call site of this method:

- There is a check in the deregistration handler for cluster existence.
- If called from within the reconciler / other handler then the check will have to be there. If it is not, this method won't fail as an update of a non existing record in DB is always successful with a mesage like `0 rows affected`.

So far this is only called from the deregistration handler. I think we are good, the less DB round trips the better.